app/Services/Sync.php

```diff
@@ -3,39 +3,210 @@
 
 use \stdClass;
 use \WP_Query;
+use \WP_CLI;
+
 use AgreableCatfishImporterPlugin\Services\Post;
+use AgreableCatfishImporterPlugin\Services\Queue;
+use AgreableCatfishImporterPlugin\Services\Worker;
+
+new Queue; // Setup Queue connection
 
 class Sync {
 
-  public static function getCategories() {
-    $site_url = get_field('catfish_website_url', 'option');
-    return Sitemap::getCategoriesFromIndex($site_url . 'sitemap-index.xml');
+  /**
+   * Queue Functions (Called from admin)
+   */
+
+  /**
+   * Queue Category / All
+   */
+  public static function queueCategory($categorySitemap = 'all', $onExistAction = 'update') {
+   // Push item into Queue
+   if(is_string(Queue::push('importCategory', array('url' => $categorySitemap, 'onExistAction' => $onExistAction)))) {
+     // Return post url
+     return $categorySitemap;
+   }
+   // If it failed return false
+   return false;
   }
 
-  public static function importCategory($categorySitemap, $limit = 10, $mostRecent = true) {
-    $postUrls = Sitemap::getPostUrlsFromCategory($categorySitemap);
-    $response = new stdClass();
-    $response->posts = [];
-    if ($limit !== -1) {
-      $postUrls = array_slice($postUrls, 0, $limit);
+  /**
+   * Queue Single URL
+   */
+  public static function queueUrl($url, $onExistAction = 'update') {
+   // Push item into Queue
+   if(is_string(Queue::push('importUrl', array('url' => $url, 'onExistAction' => $onExistAction)))) {
+     // Return post url
+     return $url;
+   }
+   // If it failed return false
+   return false;
```

Comment:
Added some clear error messages, show the error message in the errors and added a couple more try catch's to boot!